src/Caching/StackExchangeRedis/test/CacheServiceExtensionsTests.cs

```diff
@@ -142,16 +144,46 @@ public void AddStackExchangeRedisCache_HybridCacheDetected(bool hybridCacheActiv
         services.AddStackExchangeRedisCache(options => { });
         if (hybridCacheActive)
         {
-            services.TryAddSingleton<HybridCache>(new DummyHybridCache());
+            services.AddMemoryCache();
+            services.TryAddSingleton<HybridCache, DummyHybridCache>();
         }
 
         using var provider = services.BuildServiceProvider();
         var cache = Assert.IsAssignableFrom<RedisCache>(provider.GetRequiredService<IDistributedCache>());
-        Assert.Equal(hybridCacheActive, cache.HybridCacheActive);
+        Assert.Equal(hybridCacheActive, cache.IsHybridCacheActive());
     }
 
     sealed class DummyHybridCache : HybridCache
     {
+        // emulate the layout from HybridCache in dotnet/extensions
+        public DummyHybridCache(IOptions<HybridCacheOptions> options, IServiceProvider services)
+        {
+            if (services is null)
+            {
+                throw new ArgumentNullException(nameof(services));
+            }
+
+            var l1 = services.GetRequiredService<IMemoryCache>();
+            _ = options.Value;
+            var logger = services.GetService<ILoggerFactory>()?.CreateLogger(typeof(HybridCache)) ?? NullLogger.Instance;
+            // var clock = services.GetService<TimeProvider>() ?? TimeProvider.System;
+            var l2 = services.GetService<IDistributedCache>(); // note optional
+
+            // ignore L2 if it is really just the same L1, wrapped
+            // (note not just an "is" test; if someone has a custom subclass, who knows what it does?)
+            if (l2 is not null
+                && l2.GetType() == typeof(MemoryDistributedCache)
+                && l1.GetType() == typeof(MemoryCache))
+            {
+                l2 = null;
+            }
+
+            IHybridCacheSerializerFactory[] factories = services.GetServices<IHybridCacheSerializerFactory>().ToArray();
+            Array.Reverse(factories);
```

Comment:
?

Reply:
Basically: make sure we touch everything the "real" other half (in a different assembly, different repo), touches. So yes, this isn't actually useful here *other than* to assert "we've asked for all the same things"